docs/scripts/render_hubs.py

```diff
@@ -0,0 +1,110 @@
+"""Pull latest list of hubs served by infrastructure/ and save as a CSV table."""
+from pathlib import Path
+
+import pandas as pd
+from yaml import safe_load
+
+path_root = Path(__file__).parent.parent
+path_tmp = path_root / "tmp"
+path_clusters = path_root / "../config/clusters"
+
+# Grab the latest list of clusters defined in infrastructure/ explicitly ignoring
+# the test clusters in the ./tests directory
+clusters = [
+    filepath
+    for filepath in path_clusters.glob("**/*cluster.yaml")
+    if "tests/" not in str(filepath)
+]
+
+hub_list = []
+for cluster_info in clusters:
+    cluster_path = cluster_info.parent
+    if "schema" in cluster_info.name or "staff" in cluster_info.name:
+        continue
+    # For each cluster, grab it's YAML w/ the config for each hub
+    yaml = cluster_info.read_text()
+    cluster = safe_load(yaml)
+
+    # Pull support chart information to populate fields (if it exists)
+    support_files = cluster.get("support", {}).get("helm_chart_values_files", None)
+
+    # Incase we don't find any Grafana config, use an empty string as default
+    grafana_url = ""
+
+    # Try to identify the data centre location of the cluster
+    # For kubeconfig and Azure providers, this will always default to None since
+    # we do not store that information in the cluster.yaml file
+    #
+    # First try "zone" for GCP clusters
+    datacentre_loc = cluster.get(cluster["provider"], {}).get("zone", None)
+    if datacentre_loc is None:
+        # Try "region" for AWS clusters
+        datacentre_loc = cluster.get(cluster["provider"], {}).get("region", None)
+
+    # Loop through support files, look for Grafana config, and grab the URL
+    if support_files is not None:
+        for support_file in support_files:
+            with open(cluster_path.joinpath(support_file)) as f:
+                support_config = safe_load(f)
+
+            grafana_url = (
+                support_config.get("grafana", {}).get("ingress", {}).get("hosts", "")
+            )
+            # If we find a grafana url, set it here and break the loop, we are done
+            if isinstance(grafana_url, list):
+                grafana_url = grafana_url[0]
+                grafana_url = f"[{grafana_url}](http://{grafana_url})"
+                break
+
+    # Track the provider for this cluster for use later
+    provider = cluster["provider"] if cluster["provider"] != "kubeconfig" else "azure"
```

Comment:
Be careful with this. While _right now_ it's true that the only `kubeconfig` clusters we have are on Azure, we could feasibly export a `kubeconfig` for all cloud vendors and use that. It's not a robust test for "running on Azure".